### PR TITLE
feat(core/inline-idl-parser): support exceptions

### PR DESCRIPTION
**Pull request by marcoscaceres**
So now:

```
{{ "NotAllowedError" }}
```

Instead of ugly:

```
"{{ NotAllowedError }}"
```

### CODE REVIEW

File: src/core/inline-idl-parser.js [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import hyperHTML from "hyperhtml";
 import { showInlineError } from "./utils";
 
+const exceptionRegex = /\B"([^"]*)"\B/; // {{ "SomeException" }}
```
Comment (saschanaz) [inline src/core/inline-idl-parser.js]: Seems exactly same as `enumValueRegex`, are we okay with it?

Comment (marcoscaceres) [reply src/core/inline-idl-parser.js]: Yeah, it's the same... I might just kill it the enum-value one. Using `Foo["value"]` is better... alternative, is that we support both, but that's really messy.

Comment (saschanaz) [reply src/core/inline-idl-parser.js]: Might be better to remove that first, as it seems this shadows enum one...

Comment (marcoscaceres) [reply src/core/inline-idl-parser.js]: @sidvishnoi, I'm a bit confused as to how `Enum["value"]` becomes just "value" (which is good... but just can't figure it out).

Comment (marcoscaceres) [reply src/core/inline-idl-parser.js]: Never mind... figured it out 🤪

**Comment by sidvishnoi** [comment]
Just need to remove tests for `enum-value`

File: src/core/inline-idl-parser.js [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
 const methodRegex = /(\w+)\((.*)\)$/;
 const slotRegex = /^\[\[(\w+)\]\]$/;
 // matches: `value` or `[[value]]`
 // NOTE: [[value]] is actually a slot, but database has this as type="attribute"
 const attributeRegex = /^((?:\[\[)?(?:\w+)(?:\]\])?)$/;
-const enumRegex = /^(\w+)\["([\w ]+)"\]$/;
-const enumValueRegex = /\B"([^"]*)"\B/;
+const enumRegex = /^(\w+)\["([\w- ]+)"\]$/;
```
Comment (saschanaz) [inline src/core/inline-idl-parser.js]: @sidvishnoi Why are some wrapped by `\B` and others not?

Comment (sidvishnoi) [reply src/core/inline-idl-parser.js]: Interesting. 
Not sure, must have been done to "make things work" at that time.  I think we can remove `\B`

**Comment by saschanaz** [comment]
Maybe we should mark this a breaking change?

**Comment by marcoscaceres** [comment]
It's not exactly a breaking change... the "old way" is still supported, but falls back to xref's _IDL_ magic matcher (so should remain backwards compatible). The quotes makes it explicit that what's inside is an exception.

**Comment by sidvishnoi** [comment]
I think @saschanaz meant by "remove empty enum-value support" in 3057b30

**Comment by saschanaz** [comment]
Not sure I understand...

`{{ "str" }}` previous explicitly represented enum-value, now it explicitly represents exception, and the previous way to write enum-value automatically fallbacks?

> I think @saschanaz meant by "remove empty enum-value support" in [3057b30](https://github.com/w3c/respec/commit/3057b30f2034ed78f2750146f18cac48b3da3847)

Oops, didn't even notice that. How can we represent empty enum value after this PR?

**Comment by sidvishnoi** [comment]
> How can we represent empty enum value after this PR?

Linking a enum-value without the Enum context will not be supported. The syntax for enum/enum-value will be `{{ Enum["enum value"] }}` which will render as `<code><a>"enum value"</a></code>`

**Comment by marcoscaceres** [comment]
> How can we represent empty enum value after this PR?

Do you mean?: 
```
{{ SomeEnum[""] }}
```

**Comment by marcoscaceres** [comment]
> {{ "str" }} previous explicitly represented enum-value, now it explicitly represents exception, and the previous way to write enum-value automatically fallbacks?

Ah, got it. Yes, in that sense, it kinda is a breaking change. I'm conflicted, because xref and the microsyntax is still very much "beta".

File: src/core/inline-idl-parser.js (hunk 2 above)
Comment (sidvishnoi) [inline src/core/inline-idl-parser.js]:
```suggestion
const enumRegex = /^(\w+)\["([\w- ]*)"\]$/;
```

for empty enum value (missed it!)

**Comment by saschanaz** [comment]
> 
> 
> > How can we represent empty enum value after this PR?
> 
> Do you mean?:
> 
> ```
> {{ SomeEnum[""] }}
> ```

`/^(\w+)\["([\w- ]+)"\]$/.test('SomeEnum[""]') === false`, because of `+`. I think we shouldn't remove test for the empty enum value in https://github.com/w3c/respec/commit/3057b30f2034ed78f2750146f18cac48b3da3847...

**Comment by marcoscaceres** [comment]
@saschanaz, you are, of course, correct. Adding a test. I think @sidvishnoi just fixed it with his suggestion.

File: tests/spec/core/xref-spec.js [hunk 3]
```diff
@@ -763,19 +763,13 @@ describe("Core — xref", () => {
           enum Foo { "dashed-thing", "" };
           </pre>
           <p id="link1">{{ ServiceWorkerUpdateViaCache["imports"] }}</p>
-          <p id="link3"
-            data-cite="css-layout-api" data-xref-for="ChildDisplayType"
-          >{{ "block" }} {{"block"}} </p>
```
Comment (sidvishnoi) [inline tests/spec/core/xref-spec.js]: this won't be supported.

File: src/core/inline-idl-parser.js [hunk 4]
```diff
@@ -132,6 +132,7 @@ function renderEnum(details) {
     data-xref-type="enum-value"
     data-link-for="${identifier}"
     data-xref-for="${identifier}"
+    data-lt="${!enumValue ? "the-empty-string" : null}"
```
Comment (saschanaz) [inline src/core/inline-idl-parser.js]: Is this Shepherd compatible? What does Shepherd to for this?

Comment (sidvishnoi) [reply src/core/inline-idl-parser.js]: Good catch. It isn't.
Need to convert `the-empty-string` back to `""` in core/xref. 
For local linking, need to keep using `the-empty-string`, right? (I used this as we were already using it until 57d9bbc).

File: tests/spec/core/xref-spec.js [hunk 5]
```diff
@@ -762,25 +765,32 @@ describe("Core — xref", () => {
...
-      expect(link1.querySelector("a").textContent).toBe("imports");
+      expect(link1.textContent).toBe(`"no-referrer"`);
+      expect(link1.querySelector("a").textContent).toBe("no-referrer");
       expect(link1.querySelector("a").href).toBe(
-        expectedLinks.get("ServiceWorkerUpdateViaCache.imports")
+        "https://www.w3.org/TR/referrer-policy/#dom-referrerpolicy-no-referrer"
+      );
+
+      const link2 = doc.getElementById("link2");
+      // expect(link2.textContent).toBe(`""`); // TODO
```
Comment (sidvishnoi) [inline tests/spec/core/xref-spec.js]: It's coming out to be `"Referrer Policy"` 😕

Comment (marcoscaceres) [reply tests/spec/core/xref-spec.js]: magic...

**Pull request merged**